src/devices/Media/VideoDevice/Devices/UnixVideoDevice.cs

```diff
@@ -0,0 +1,509 @@
+            }
+
+            if (Settings.Brightness.Equals(default))
+            {
+                Settings.Brightness = GetVideoDeviceValue(VideoDeviceValueType.Brightness).DefaultValue;
+            }
+
+            if (Settings.Saturation.Equals(default))
+            {
+                Settings.Saturation = GetVideoDeviceValue(VideoDeviceValueType.Saturation).DefaultValue;
+            }
+
+            if (Settings.Sharpness.Equals(default))
+            {
+                Settings.Sharpness = GetVideoDeviceValue(VideoDeviceValueType.Sharpness).DefaultValue;
+            }
+
+            if (Settings.Contrast.Equals(default))
+            {
+                Settings.Contrast = GetVideoDeviceValue(VideoDeviceValueType.Contrast).DefaultValue;
+            }
+
+            if (Settings.Gain.Equals(default))
+            {
+                Settings.Gain = GetVideoDeviceValue(VideoDeviceValueType.Gain).DefaultValue;
+            }
+
+            if (Settings.Gamma.Equals(default))
+            {
+                Settings.Gamma = GetVideoDeviceValue(VideoDeviceValueType.Gamma).DefaultValue;
+            }
+
+            if (Settings.Rotate.Equals(default))
+            {
+                Settings.Rotate = GetVideoDeviceValue(VideoDeviceValueType.Rotate).DefaultValue;
+            }
+
+            if (Settings.WhiteBalanceTemperature.Equals(default))
+            {
+                Settings.WhiteBalanceTemperature = GetVideoDeviceValue(VideoDeviceValueType.WhiteBalanceTemperature).DefaultValue;
+            }
+
+            if (Settings.ColorEffect.Equals(default))
+            {
+                Settings.ColorEffect = (ColorEffect)GetVideoDeviceValue(VideoDeviceValueType.ColorEffect).DefaultValue;
+            }
+
+            if (Settings.PowerLineFrequency.Equals(default))
+            {
+                Settings.PowerLineFrequency = (PowerLineFrequency)GetVideoDeviceValue(VideoDeviceValueType.PowerLineFrequency).DefaultValue;
+            }
+
+            if (Settings.SceneMode.Equals(default))
+            {
+                Settings.SceneMode = (SceneMode)GetVideoDeviceValue(VideoDeviceValueType.SceneMode).DefaultValue;
+            }
+
+            if (Settings.WhiteBalanceEffect.Equals(default))
+            {
+                Settings.WhiteBalanceEffect = (WhiteBalanceEffect)GetVideoDeviceValue(VideoDeviceValueType.WhiteBalanceEffect).DefaultValue;
+            }
+
+            if (Settings.HorizontalFlip.Equals(default))
+            {
+                Settings.HorizontalFlip = Convert.ToBoolean(GetVideoDeviceValue(VideoDeviceValueType.HorizontalFlip).DefaultValue);
+            }
+
+            if (Settings.VerticalFlip.Equals(default))
+            {
+                Settings.VerticalFlip = Convert.ToBoolean(GetVideoDeviceValue(VideoDeviceValueType.VerticalFlip).DefaultValue);
+            }
+        }
+
+        private void Initialize()
```

Comment:
can you just always initialize in the constructor and remove the lock at all?

Reply:
If I change the shooting parameters after the first capture, then the second capture will throw an error. So the process must be initialization -> shooting -> close.